prow/plugins/transfer-issue/transfer-issue.go

```diff
@@ -0,0 +1,110 @@
+package transferissue
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "transfer-issue"
+
+var (
+	transferRe = regexp.MustCompile(`(?mi)^/transfer(?:-issue)?\s*(.*)$`)
+)
+
+type githubClient interface {
+	GetRepo(org, name string) (github.FullRepo, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsCollaborator(org, repo, user string) (bool, error)
+	TransferIssue(org, dstRepoNodeID string, issueNodeID string) (*github.TransferIssueMutation, error)
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The transfer-issue plugin transfers a GitHub issue from one repo to another in the same organization.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/transfer[-issue] <destination repo in same org>",
+		Description: "Transfers an issue to a different repo in the same org.",
+		Featured:    true,
+		WhoCanUse:   "Org members.",
+		Examples:    []string{"/transfer-issue kubectl", "/transfer test-infra"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handleTransfer(pc.GitHubClient, pc.Logger, e)
+}
+
+func handleTransfer(gc githubClient, log *logrus.Entry, e github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	srcRepoName := e.Repo.Name
+	srcRepoPair := org + "/" + srcRepoName
+	user := e.User.Login
+
+	if e.IsPR || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	matches := transferRe.FindAllStringSubmatch(e.Body, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+	if len(matches) != 1 || len(matches[0]) != 2 {
+		log.Warnf("invalid usage: %v", matches)
+		return gc.CreateComment(
+			org, srcRepoName, e.Number,
+			plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, "/transfer-issue must only be used once and with a single destination repo."),
+		)
+	}
+
+	dstRepoName := matches[0][1]
+	dstRepoPair := org + "/" + dstRepoName
+
+	dstRepo, err := gc.GetRepo(org, dstRepoName)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch destination repo: %s", dstRepoPair)
+		// TODO: Might want to add another GetRepo type call that checks if a repo exists vs a bad request
+		return gc.CreateComment(
+			org, srcRepoName, e.Number,
+			plugins.FormatResponseRaw(e.Body, e.HTMLURL, user, fmt.Sprintf("Something went wrong or the destination repo %s does not exist.", dstRepoPair)),
+		)
+	}
+
+	isCollaboratorSrc, err := gc.IsCollaborator(org, srcRepoName, user)
```

Comment:
We generally use this check to determine if we trust a user: https://github.com/kubernetes/test-infra/blob/0757032ed149f086f3eb69efe20142c75b371450/prow/plugins/welcome/welcome.go#L122

Reply:
I can switch to that but there's a mix of different usages. Is `trigger.TrustedUser` the way forward? If so I'll open an issue to track that refactor.

[lgtm: IsCollaborator](https://github.com/kubernetes/test-infra/blob/7af3c135a4cceb136a71516a86f2af94e030c86c/prow/plugins/lgtm/lgtm.go#L283)

[label: IsMember](https://github.com/kubernetes/test-infra/blob/7af3c135a4cceb136a71516a86f2af94e030c86c/prow/plugins/label/label.go#L190)

[lifecycle: IsCollaborator](https://github.com/kubernetes/test-infra/blob/7af3c135a4cceb136a71516a86f2af94e030c86c/prow/plugins/lifecycle/reopen.go#L54)

[releasenote: IsMember](https://github.com/kubernetes/test-infra/blob/7af3c135a4cceb136a71516a86f2af94e030c86c/prow/plugins/releasenote/releasenote.go#L138)

[sigmention: IsMember](https://github.com/kubernetes/test-infra/blob/7af3c135a4cceb136a71516a86f2af94e030c86c/prow/plugins/sigmention/sigmention.go#L150)

Reply:
Hm good point, so `trigger.TrustedUser` is for running tests which is not necessarily the same trust level we want for this. So not sure. Note that `IsCollaborator` is the widest possible set, as it includes both org members and ppl that are  collaborators on the repo itself.